packages/react/src/create-strict-api/__tests__/index.test.tsx

```diff
@@ -0,0 +1,310 @@
+    });
+
+    it('should type error for unsupported known pseudos', () => {
+      function Button({ xcss }: { xcss: ReturnType<typeof XCSSProp<'background', never>> }) {
+        return <button data-testid="button" className={xcss} />;
+      }
+      const { getByTestId } = render(
+        <Button
+          xcss={{
+            // @ts-expect-error — Object literal may only specify known properties, and '':hover'' does not exist in type
+            ':hover': {
+              color: 'red',
+            },
+          }}
+        />
+      );
+
+      expect(getByTestId('button')).toHaveCompiledCss('color', 'red', { target: ':hover' });
+    });
+
+    it('should type error for unsupported unknown pseudos', () => {
+      function Button({ xcss }: { xcss: ReturnType<typeof XCSSProp<'background', never>> }) {
+        return <button data-testid="button" className={xcss} />;
+      }
+
+      const { getByTestId } = render(
+        <Button
+          xcss={{
+            // @ts-expect-error — Object literal may only specify known properties, and '':hover'' does not exist in type
+            ':asd': {
+              color: 'red',
+            },
+          }}
+        />
+      );
+
+      expect(getByTestId('button')).toHaveCompiledCss('color', 'red', { target: ':asd' });
+    });
+
+    it('should type error for invalid known values in pseudos', () => {
+      function Button({ xcss }: { xcss: ReturnType<typeof XCSSProp<'color', '&:hover'>> }) {
+        return <button data-testid="button" className={xcss} />;
+      }
+
+      const { getByTestId } = render(
+        <Button
+          xcss={{
+            '&:hover': {
+              // @ts-expect-error — Type '"red"' is not assignable to type 'CompiledPropertyDeclarationReference | "var(--ds-text)" | undefined'.ts(2322)
+              color: 'red',
+            },
+          }}
+        />
+      );
+
+      expect(getByTestId('button')).toHaveCompiledCss('color', 'red', { target: ':hover' });
+    });
+
+    it('should type error for invalid unknown values in pseudos', () => {
+      function Button({ xcss }: { xcss: ReturnType<typeof XCSSProp<'color', '&:hover'>> }) {
+        return <button data-testid="button" className={xcss} />;
+      }
+
+      const { getByTestId } = render(
+        <Button
+          xcss={{
+            '&:hover': {
+              // @ts-expect-error — Type '{ asd: string; }' is not assignable to type 'MarkAsRequired<XCSSItem<"color", { color: "var(--ds-text)"; }>, never>'.
+              asd: 'red',
+            },
+          }}
+        />
+      );
+
+      expect(getByTestId('button')).toHaveCompiledCss('asd', 'red', { target: ':hover' });
+    });
+
+    it('should enforce required properties', () => {
+      function Button({
+        xcss,
+      }: {
+        xcss: ReturnType<
+          typeof XCSSProp<
+            'background',
+            never,
+            { requiredProperties: 'background'; requiredPseudos: never }
```

Comment:
I find this argument a bit strange, can't consumers construct it instead using combinations of `Partial` / `Required` etc?

In this model, would it be possible to do something like this? I haven't looked at the return type too closely

```
// assume default is partial, but it's based on the schema really
ReturnType<typeof XCSSProp<'background', never>>
  & Required<ReturnType<typeof XCSSProp<'color', never>>>
```

Reply:
We technically could, but this way allows it to be:

1. type safe, it has to be a value passed into the generic 1st/2nd args
2. easier, less of a headache to type out / ensure it's correct

I'm for keeping it as is.